Testscripts/Linux/XDP-Action.sh

```diff
@@ -0,0 +1,137 @@
+# Licensed under the Apache License.
+
+# This script verifies XDP Action - TX or DROP
+# For DROP Action verification script will compare number of ping packets
+#		lossed before XDP DROP action and after XDP DROP action
+# For TX Action verification script will start ping on server and measure the packets captured by tcpdump
+#		Expected result: 0 packets should be captured by tcpdump after TX action as this action
+#		will return all ping packets to sender at hardware level
+
+
+# Helper function
+# This function will start xdpdump on client 
+# and ping from server on eth1 network interface
+function ping_test () {
+	local logSuffix=$1
+
+	xdpdumpCommand="cd bpf-samples/xdpdump && timeout 20 ./xdpdump -i ${nicName} > ~/xdpdumpout_${logSuffix}.txt"
+	LogMsg "Starting xdpdump on client with command: ${xdpdumpCommand}"
+	ssh ${client} "sh -c '${xdpdumpCommand} &'"
+
+	LogMsg "Starting ping on server with command: ping -I ${nicName} -c 10 ${clientSecondIP} > ~/pingOut_${logSuffix}.txt"
+	ssh ${server} "ping -I ${nicName} -c 10 ${clientSecondIP} > ~/pingOut_${logSuffix}.txt"
+}
+
+# Helper function
+# This function will start tcpdump on client and start ping_test
+function ping_with_tcpdump () {
+	local logSuffix=$1
+	LogMsg "Ips client $client server $server"
+	# Start tcpdump
+	LogMsg "Starting tcpdump on $client"
+	ssh ${client} "sh -c 'timeout 10 tcpdump -i eth1 icmp -w tcpdumptest_${logSuffix}.pcap &'"
+	ping_test $logSuffix
+}
+
+UTIL_FILE="./utils.sh"
+
+# Source utils.sh
+. ${UTIL_FILE} || {
+		echo "ERROR: unable to source ${UTIL_FILE}!"
+		echo "TestAborted" > state.txt
+		exit 0
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+
+# Script start from here
+LogMsg "*********INFO: Script Execution Started********"
+
+# check for client and server ips are present
+for ip in ${client} ${server}
+do
+	CheckIP ${ip}
+	if [ $? -eq 1 ]; then
+		LogErr "ERROR: Please provide valide client and server ip. Invalide ip: ${ip}"
+		SetTestStateAborted
+		exit 0
+	fi
+done
+
+if [ "${ACTION}" == "DROP" ];then
+	LogMsg "Initializing validation for XDP Action ${ACTION}"
+
+	# Ping test before changing action
+	ping_test "before"
+	check_exit_status "ping test before drop config"
+
+	# Build xdpdump with DROP Config
+	ssh ${client} "cd bpf-samples/xdpdump && make clean && export CFLAGS='-D __ACTION_DROP__ -I../libbpf/src/root/usr/include' && make"
+	check_exit_status "Building xdpdump with DROP config"
+
+	# Ping test after changing action
+	ping_test "after"
+	check_exit_status "ping test after drop config"
+
+	LogMsg "Starting analysis on number of ping packets drop.."
+	packetDropBefore=$( ssh ${server} "grep -oP '\d+(?=% packet loss)' ~/pingOut_before.txt" )
+	LogMsg "Packet drop before $packetDropBefore"
+	packetDropAfter=$( ssh ${server} "grep -oP '\d+(?=% packet loss)' ~/pingOut_after.txt" )
+	LogMsg "Packet drop after $packetDropAfter"
+	
+	if [ $((packetDropBefore + 30)) -lt $((packetDropAfter)) ];then
```

Comment:
30?

Reply:
Numbers that I have been receiving while testing are packetDropBefore = 0 packetDropAfter=100.
I have kept a 30% buffer for packet loss due to network.
@haiyangz please let me know your opinion on this.

Reply:
Made a constant for pakcetLossInNetwork = 10.